game.sy

```diff
@@ -99,8 +99,75 @@ player_new :: fn pos: (float, float) -> Player {
 
 :Player is :Entity
 
+Puck :: blob {
+    pos: (float, float),
+    rot: float,
+    vel: float,
+    acc: float,
+    holder: Player | void,
+    hold_cooldown: float,
+    draw: fn -> void,
+    process_input: fn Input, float -> void,
+    update: fn float -> void,
+}
+
+puck_new :: fn pos: (float, float) -> Puck {
+    self: !Puck = nil
+    self = Puck{
+        pos: pos,
+        rot: 0.0,
+        vel: 0.0,
+        acc: 0.0,
+        holder: nil,
+        hold_cooldown: 0.0,
+        draw: fn -> void {
+            rgb :: (1.0, 1.0, 1.0)
+            l_gfx_rect' self.pos, (0.04, 0.04), self.rot, rgb
+            l_gfx_rect' self.pos, (0.04, 0.04), self.rot + 3.14 / 3.0, rgb
+            l_gfx_rect' self.pos, (0.04, 0.04), self.rot + 3.14 * 2.0 / 3.0, rgb
+        },
+        process_input: fn input: Input, delta: float -> void {
+            if input.shoot && self.holder is :Player {
+                holder: !Player = self.holder
+                self.vel = holder.vel * 2.0
+                self.rot = holder.rot
+                self.holder = nil
+                self.hold_cooldown = 0.1
```

Comment:
Shouldn't this logic be placed on the player? Do we really want to give input to the puck?

Reply:
Yes, I am not sure what I was thinking